scripts/prepare-crds

```diff
@@ -56,16 +59,40 @@ for crd_yaml in ${f}/charts-crd/templates/*; do
 	crd_apis+=("${crd_group}/${crd_version}/${crd_kind}")
 done
 
-# Init Chart.yaml for CRD chart
-cat << EOF > ${f}/charts-crd/Chart.yaml
-apiVersion: ${api_version}
-version: ${chart_version}
-description: A Rancher chart that creates ${name} CRDs within a cluster.
-name: ${name}-crd
-type: application
-EOF
+# Set environment variable to be used in validate-install-crd.yaml template
+set_found_crd=$(
+for crd in ${crd_apis[@]}; do
+  echo "# {{- set \$found \"${crd}\" false -}}"
+done
+)
 
-# Add providesGVR annotation
+# Copy base template and apply variables to the template
+mkdir -p ${f}/charts-crd
+cp -R ./chart-templates/crd-base/* ${f}/charts-crd
+assume_ownership=$(cat ${f}/package.yaml | yq r - generateCRDChart.assumeOwnershipOfCRDs)
+if [[ ${assume_ownership} == "true" ]]; then
+    cp -fR ./chart-templates/crd-assume-ownership/* ${f}/charts-crd
+fi
+for template_file in $(find ${f}/charts-crd -type f); do
+  # Applies any environment variables currently set onto your template file
+  eval "echo \"$(sed 's/"/\\"/g' ${template_file})\"" > ${template_file}
```

Comment:
This is basically a templating function ?

Reply:
Yes, this command basically runs `eval` on the contents of the template file, which automatically subs out the global variables. The `sed` command within it just ensures that double quotes don't get lost in translation